nodeJS/orms/prisma_orm.md

```diff
@@ -0,0 +1,157 @@
+
+</div>
+
+#### Navigating the codebase
+
+When all database interactions are done in raw SQL, there's nowhere in the codebase where you can understand the database tables, their relations, and column data types. You might have to log in to your database to make sense of what the codebase is doing. To attain a technical understanding of the project, you now rely on the codebase as well as access to the database.
+
+Most ORMs out there solve this problem by bringing database definitions into the codebase. This is called a "schema". This allows you to quickly glance at the schema of a table and understand what columns it has and so forth.
+
+#### Altering production data
+
+Databases inevitably change as the requirements of the project evolve. You might have to add a new column or populate a new table with existing data. In technical terms, this is called a migration. Without an ORM or a similar library, you'll be hand rolling these migrations, which will be error-prone and tedious. ORMs standardize migrations via changelogs, and have processes to deal with conflicts. Granted, you won't be running as many migrations in your curriculum projects, but as you work professionally, you might do it every other day.
+
+### Introducing Prisma ORM
+
+ORMs pretty much solve all of the pain points mentioned above but it's not all sunshine and roses. There can be a learning curve to fully understand the ins and outs of an ORM and some don't even fully support all SQL features. Even with these shortcomings, using an ORM can be extremely worthwhile.
+
+In the Node.js ecosystem, there are A LOT of ORMs to choose from, and the community has not landed on a go-to yet. We've decided to cover [Prisma](https://www.prisma.io/) in the curriculum because of its popularity and community support.
+
+Prisma has all the features we need to complete this curriculum and then some. Prisma is a library, which means you can just use `npm install` to use it. We'll start by discussing some of its features, and then link to some official Prisma guides to help you get started.
+
+### Prisma Schema
+
+The Prisma schema is a file where you will define your models. For example, consider a message table in a chat app:
+
+```text
+model Message {
+   id        Int      @id @default(autoincrement())
+   content   String   @db.VarChar(255) 
+   createdAt DateTime @default(now())
+   author    User     @relation(fields: [authorId], references: [id])
+   authorId  Int     
+}
+
+model User {
+   // user's fields
+}
+```
+
+There's a bunch of new stuff up here. Prisma schema files are written in Prisma Schema Language. You can see how not only there are column definitions for the table, but also a relation to another table defined within the `Message` model. You'll learn more about this in the assignment readings.
+
+This schema file lives in your codebase and is tracked by version control. You can guess by now how useful this is.
```

Comment:
I think it might be good to mention what the schema file is (that it has a specific naming convention) and that there can be multiple schema files in Prisma. e.g. from the docs:
> It is typically a single file called schema.prisma (or multiple files with .prisma file extension) that is stored in a defined but customizable [location](https://www.prisma.io/docs/orm/prisma-schema/overview/location).

https://www.prisma.io/docs/orm/prisma-schema/overview

Reply:
I purposefully kept out this information, since learners will only ever need one schema file for the projects they tackle in the curriculum. They do learn about the naming convention and the multiple files stuff in the linked documentation in the assignment section.